src/explore-education-statistics-admin/src/pages/release/data/components/__tests__/DataFileReplacementPlan.test.tsx

```diff
@@ -367,308 +381,369 @@ describe('DataReplacementPlan', () => {
+        filtersComplete: true,
+      },
+      valid: true,
+    },
     valid: true,
   };
 
-  test('renders correctly with invalid plan', async () => {
-    dataReplacementService.getReplacementPlan.mockResolvedValue(
-      testReplacementPlan,
-    );
-
-    render(
-      <MemoryRouter>
-        <DataFileReplacementPlan
-          cancelButton={<button type="button">Cancel</button>}
-          publicationId="publication-1"
-          releaseVersionId="release-1"
-          fileId="file-1"
-          replacementFileId="file-2"
-        />
-      </MemoryRouter>,
-    );
-
-    await waitFor(() => {
-      expect(screen.getByText('Data blocks: ERROR')).toBeInTheDocument();
-      expect(screen.getByText('Footnotes: ERROR')).toBeInTheDocument();
-    });
-
-    expect(
-      screen.getByText(
-        /One or more data blocks will be invalidated by this data replacement/,
-      ),
-    ).toBeInTheDocument();
-    expect(
-      screen.getByText(
-        /One or more footnotes will be invalidated by this data replacement/,
-      ),
-    ).toBeInTheDocument();
-
-    const details = screen.getAllByRole('group');
-
-    expect(details).toHaveLength(4);
-
-    const dataBlock1 = within(details[0]);
-
-    expect(
-      dataBlock1.getByRole('button', { name: /Data block 1/ }),
-    ).toHaveTextContent('ERROR');
-
-    expect(dataBlock1.getByTestId('Country')).toHaveTextContent('England');
-    expect(dataBlock1.getByTestId('Country')).not.toHaveTextContent(
-      'England not present',
-    );
-
-    expect(dataBlock1.getByTestId('Local Authority')).toHaveTextContent(
-      'Barnsley',
-    );
-    expect(dataBlock1.getByTestId('Local Authority')).not.toHaveTextContent(
-      'Barnsley not present',
-    );
-
-    expect(dataBlock1.getByTestId('Local Authority')).toHaveTextContent(
-      'Birmingham not present',
-    );
-    expect(dataBlock1.getByTestId('Start date')).toHaveTextContent(
-      '2016 not present',
-    );
-
-    expect(dataBlock1.getByTestId('End date')).toHaveTextContent('2018');
-    expect(dataBlock1.getByTestId('End date')).not.toHaveTextContent(
-      '2018 not present',
-    );
-
-    expect(dataBlock1.getByTestId('Filter 1')).toHaveTextContent('Group 1');
-    expect(dataBlock1.getByTestId('Filter 1')).toHaveTextContent('Group 2');
-
-    expect(dataBlock1.getByTestId('Group 1')).toHaveTextContent(
-      'Item 1 not present',
-    );
-    expect(dataBlock1.getByTestId('Group 1')).toHaveTextContent('Item 2');
-    expect(dataBlock1.getByTestId('Group 1')).not.toHaveTextContent(
-      'Item 2 not present',
-    );
-
-    expect(dataBlock1.getByTestId('Group 2')).toHaveTextContent('Item 3');
-    expect(dataBlock1.getByTestId('Group 2')).not.toHaveTextContent(
-      'Item 3 not present',
-    );
-
-    expect(dataBlock1.getByTestId('Filter 2')).toHaveTextContent('Group 3');
-    expect(dataBlock1.getByTestId('Filter 2')).toHaveTextContent('Group 4');
+  test.each([[true], [false]])(
```

Comment:
Personally, I feel this would be more simple with 2 separate tests, one with the feature enabled and one without with each to have distinct test titles, otherwise it might be confusing when we're getting failures

```ts
describe('renders correctly with invalid plan', ()=>{
  test('with ReplacementOfPublicApiDataSets enabled', async () => { /* ... */});
  test('with ReplacementOfPublicApiDataSets disabled', async () => { /* ... */});
})
```